plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSTypeValidator.xtend

```diff
@@ -518,27 +520,54 @@ class N4JSTypeValidator extends AbstractN4JSDeclarativeValidator {
 				return;
 			}
 
+			val usedInSpecConstructor = objectLiteral.eContainer?.eContainer instanceof NewExpression;
 			val ctor = containerTypesHelper.fromContext(objectLiteral).findConstructor(type as ContainerType<?>);
 			val strategyFilter = new TypingStrategyFilter(typingStrategy,
 				typingStrategy === TypingStrategy.STRUCTURAL_WRITE_ONLY_FIELDS,
 				isSpecArgumentToSpecCtor(objectLiteral, ctor));
+			val strategyFilterIncludeNotAccessible = new TypingStrategyFilter(typingStrategy,
+				typingStrategy === TypingStrategy.STRUCTURAL_WRITE_ONLY_FIELDS,
+				isSpecArgumentToSpecCtor(objectLiteral, ctor), true);
 			val expectedMembers = containerTypesHelper.fromContext(objectLiteral).allMembers(
-				type as ContainerType<?>).filter[member|strategyFilter.apply(member)].map[member|member.name].
-				toSet();
+				type as ContainerType<?>).filter[member|strategyFilter.apply(member)].map[member|member.name].toSet();
+			val expectedMembersPlusNotAccessibles = containerTypesHelper.fromContext(objectLiteral).allMembers(
+				type as ContainerType<?>).filter[member|strategyFilterIncludeNotAccessible.apply(member)].map[member|member.name].toSet();
 
 			// These are available via the 'with' keyword add them to the accepted ones
 			typeRef.structuralMembers.forEach[member|expectedMembers.add(member.name)];
 
+			var EObject container = objectLiteral;
+			while (container instanceof ObjectLiteral || container instanceof ArrayLiteral || container instanceof ArrayElement) {
+				container = container.eContainer;
+			}
+			val lhsName = switch (container) {
+				VariableDeclaration: container.name
+				AssignmentExpression: {
+					val lhs = container.lhs;
+					switch (lhs) {
+						IdentifierRef: lhs.idAsText
+						ParameterizedPropertyAccessExpression: lhs.propertyAsText
+						default: "the receiving object"
+					}
+				}
+				Argument: "the receiving parameter"
+				default: "the receiving object"
```

Comment:
In the fall back case you could consider a wording like:
`propUnknown is not defined in ~C; it will not be accessible via type ~C.`

Maybe that's more clear than "the receiving object"? (not sure)